src/set.c

```diff
@@ -561,36 +559,26 @@ static Obj FuncUNITE_SET(Obj self, Obj set1, Obj set2)
         e1 = ELM_PLIST( set1, i1 );
         e2 = ELM_PLIST( set2, i2 );
         if ( EQ( e1, e2 ) ) {
-            lenr++;
-            SET_ELM_PLIST( TmpUnion, lenr, e1 );
-            CHANGED_BAG( TmpUnion );
+            PushPlist( TmpUnion, e1 );
             i1++;  i2++;
         }
         else if ( LT( e1, e2 ) ) {
-            lenr++;
-            SET_ELM_PLIST( TmpUnion, lenr, e1 );
-            CHANGED_BAG( TmpUnion );
+            PushPlist( TmpUnion, e1 );
             i1++;
         }
         else {
-            lenr++;
-            SET_ELM_PLIST( TmpUnion, lenr, e2 );
-            CHANGED_BAG( TmpUnion );
+            PushPlist( TmpUnion, e2 );
             i2++;
         }
     }
     while ( i1 <= len1 ) {
         e1 = ELM_PLIST( set1, i1 );
```

Comment:
How about
```suggestion
         e1 = ELM_PLIST( set1, i1++ );
```
and then removing the separate `i1++` below (and similarly for the next block of code)? Just an idea.

Reply:
well, if we change this, wouldn't it be even more natural to switch from a `while` loop to a `for` loop, like `for ( ; i1 <= len1; i1++ ) {`?

Reply:
Oh. Of course! Silly me. Do so if you feel like it.